canary/webrtc-c/src/media-server-storage/kvsWebRTCClientMaster.cpp

```diff
@@ -0,0 +1,367 @@
+    DLOGI("[KVS Master] Streaming session terminated");
+
+CleanUp:
+
+    if (retStatus != STATUS_SUCCESS) {
+        DLOGE("[KVS Master] Terminated with status code 0x%08x", retStatus);
+    }
+
+    DLOGI("Exiting with 0x%08x", retStatus);
+    if (initialized) {
+        Canary::Cloudwatch::getInstance().monitoring.pushExitStatus(retStatus);
+    }
+    Canary::Cloudwatch::deinit();
+
+    DLOGI("[KVS Master] Cleaning up....");
+    if (pSampleConfiguration != NULL) {
+        // Kick of the termination sequence
+        ATOMIC_STORE_BOOL(&pSampleConfiguration->appTerminateFlag, TRUE);
+
+        if (pSampleConfiguration->mediaSenderTid != INVALID_TID_VALUE) {
+            THREAD_JOIN(pSampleConfiguration->mediaSenderTid, NULL);
+        }
+
+        retStatus = signalingClientGetMetrics(pSampleConfiguration->signalingClientHandle, &signalingClientMetrics);
+        if (retStatus == STATUS_SUCCESS) {
+            logSignalingClientStats(&signalingClientMetrics);
+        } else {
+            DLOGE("[KVS Master] signalingClientGetMetrics() operation returned status code: 0x%08x", retStatus);
+        }
+        retStatus = freeSignalingClient(&pSampleConfiguration->signalingClientHandle);
+        if (retStatus != STATUS_SUCCESS) {
+            DLOGE("[KVS Master] freeSignalingClient(): operation returned status code: 0x%08x", retStatus);
+        }
+
+        retStatus = freeSampleConfiguration(&pSampleConfiguration);
+        if (retStatus != STATUS_SUCCESS) {
+            DLOGE("[KVS Master] freeSampleConfiguration(): operation returned status code: 0x%08x", retStatus);
+        }
+    }
+    DLOGI("[KVS Master] Cleanup done");
+    CHK_LOG_ERR(retStatus);
+
+    RESET_INSTRUMENTED_ALLOCATORS();
+
+    // https://www.gnu.org/software/libc/manual/html_node/Exit-Status.html
+    // We can only return with 0 - 127. Some platforms treat exit code >= 128
+    // to be a success code, which might give an unintended behaviour.
+    // Some platforms also treat 1 or 0 differently, so it's better to use
+    // EXIT_FAILURE and EXIT_SUCCESS macros for portability.
+    return STATUS_FAILED(retStatus) ? EXIT_FAILURE : EXIT_SUCCESS;
+}
+
+STATUS readFrameFromDisk(PBYTE pFrame, PUINT32 pSize, PCHAR frameFilePath)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT64 size = 0;
+    CHK_ERR(pSize != NULL, STATUS_NULL_ARG, "[KVS Master] Invalid file size");
+    size = *pSize;
+    // Get the size and read into frame
+    CHK_STATUS(readFile(frameFilePath, TRUE, pFrame, &size));
+CleanUp:
+
+    if (pSize != NULL) {
+        *pSize = (UINT32) size;
+    }
+
+    return retStatus;
+}
+
+// Save first-frame-sent time to file for consumer-end access.
+PVOID writeFirstFrameSentTimeToFile(){
+    DLOGI("[Canary] Opening toConsumer.txt");
+    FILE *toConsumer = FOPEN("../toConsumer.txt", "w");
+    if (toConsumer == NULL)
+    {
+        printf("[Canary] Error opening toConsumer.txt\n");
+        exit(1);
```

Comment:
Not sure if we want to use exit(0) here. Also, why is this function a PVOID()? Dont see this being run in a thread.